src/mpl2/test/cpp/TestSnapper.cpp

```diff
@@ -57,16 +57,155 @@ TEST_F(Mpl2SnapperTest, CanSetMacroForEmptyInstances)
   odb::dbBlock* block_ = odb::dbBlock::create(chip_, "simple_block");
   block_->setDieArea(odb::Rect(0, 0, 1000, 1000));
 
-  odb::dbDatabase::beginEco(block_);
   odb::dbInst* inst1 = odb::dbInst::create(block_, master_, "cells_1");
   odb::dbInst* inst2 = odb::dbInst::create(block_, master_, "cells_2");
   odb::dbInst* inst3 = odb::dbInst::create(block_, master_, "cells_3");
-  odb::dbDatabase::endEco(block_);
 
   Snapper snapper;
   snapper.setMacro(inst1);
   snapper.setMacro(inst2);
   snapper.setMacro(inst3);
-}
+}  // CanSetMacroForEmptyInstances
+
+TEST_F(Mpl2SnapperTest, CanSnapMacros)
+{
+  // This test checks the snapMacro functionality of the
+  // Snapper class defined in hier_rtlmp.cpp.
+  //
+  // snapMacro works by computing a valid origin for a
```

Comment:
You're right, the wording isn't right. I've edited it to the following; does this work?
```
snapMacro works by adjusting the position of the
macro so that the signal pins are aligned with the
track-grid.

Therefore, this test inputs instances that have
not been aligned correctly as testcases, and then
checks whether snapMacro has aligned said instances
correctly.
...

At the moment, this test only receives instance unaligned with
the track-grid as inputs, because right now, Snapper will not
necessarily compute the most optimal origin for instances that
are  already aligned (origins should've been left unchanged).
```